fast64_internal/f3d_material_converter.py

```diff
@@ -77,35 +79,95 @@ def getV4PresetName(name):
     return newName
 
 
-def convertF3DtoNewVersion(obj, index, material, materialDict, version):
+def get_group_from_polygon(obj: bpy.types.Object, polygon: bpy.types.MeshPolygon):
+    sample_vert: bpy.types.MeshVertex = obj.data.vertices[polygon.vertices[0]]
+    if len(sample_vert.groups):
+        g: bpy.types.VertexGroupElement = None
+        for g in sample_vert.groups:
+            if g.weight > 0.99:
+                return obj.vertex_groups[sample_vert.groups[0].group]
+    return None
 
-    if material.mat_ver > 3:
-        oldPreset = AddPresetBase.as_filename(material.f3d_mat.presetName)
-    else:
-        oldPreset = material.f3d_preset
 
-    if version > 3:
+def has_valid_mat_ver(material: bpy.types.Material):
+    return getattr(material, "mat_ver", -1) >= 1
+
+
+def get_best_draw_layer_for_materials():
```

Comment:
I'm assuming this code is for handling old materials where the draw layer was stored in the bone/object? Maybe the function name could be something more informative like:
```suggestion
def set_draw_layer_for_old_materials():
```

Reply:
good point! just updated it and pushed